bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartWebSocket.java

```diff
@@ -139,18 +136,29 @@ public void onFrame(Frame pong) {
     @OnWebSocketClose
     public void onClose(int statusCode, String reason) {
         if (!closing) {
-            logger.debug("Connection to Gardena Webservice was closed ({}): code: {}, reason: {}", socketId, statusCode,
+            logger.warn("Connection to Gardena Webservice was closed ({}): code: {}, reason: {}", socketId, statusCode,
                     reason);
-            socketEventListener.onWebSocketClose();
+
+            // let listener handle restart of socket
+            socketEventListener.onWebSocketClose(locationID);
         }
     }
 
     @OnWebSocketError
     public void onError(Throwable cause) {
         if (!closing) {
-            logger.warn("Gardena Webservice error ({}): {}, restarting", socketId, cause.getMessage());
-            logger.debug("{}", cause.getMessage(), cause);
-            socketEventListener.onWebSocketError();
+            if (logger.isDebugEnabled()) {
```

Comment:
Why do you switch on logger.isDebugEnabled()?

Reply:
I use `if (logger.isDebugEnabled())` here to print out the whole stacktrace in case the log level is '**DEBUG**'. Otherwise I only print the short error message.

If you want, I can add an additional `logger.debug("{}", cause.getMessage(), cause);` to print the stacktrace after the other logging output. (As it was implemented before my changes)

Reply:
I think that would be easier to follow.